request-manager/server/server.go

```diff
@@ -141,20 +218,36 @@ func (s *Server) Boot() error {
 	s.appCtx.JLS = joblog.NewStore(dbc)
 
 	// Auth Manager: request authorization (pre- (built-in) and post- using plugin)
-	s.appCtx.Auth = auth.NewManager(s.appCtx.Plugins.Auth, MapACL(specs), cfg.Auth.AdminRoles, cfg.Auth.Strict)
+	s.appCtx.Auth = auth.NewManager(s.appCtx.Plugins.Auth, mapACL(specs), cfg.Auth.AdminRoles, cfg.Auth.Strict)
 
 	// API: endpoints and controllers, also handles auth via auth plugin
 	s.api = api.NewAPI(s.appCtx)
 
 	return nil
 }
 
+// API returns the Request Manager API created in Boot.
 func (s *Server) API() *api.API {
 	return s.api
 }
 
+// --------------------------------------------------------------------------
+
+// Catch TERM and INT signals to gracefully shut down the Request Manager
+func (s *Server) waitForShutdown() {
+	sigChan := make(chan os.Signal, 1)
+	signal.Notify(sigChan, syscall.SIGINT, syscall.SIGTERM)
+
+	<-sigChan
+
+	err := s.Stop()
+	if err != nil {
+		log.Errorf("error shutting down server: %s", err)
+	}
+}
+
 // MapACL maps spec file ACL to auth.ACL structure.
-func MapACL(specs grapher.Config) map[string][]auth.ACL {
+func mapACL(specs grapher.Config) map[string][]auth.ACL {
```

Comment:
Why private? I think there were unit tests for this.

Reply:
If no tests then I had intended to test it but didn’t actually do it. It’s fine for now. It needs to be refactored/moved anyway.